sdk/core/utils/redisclient/redisclient_test.go

```
@@ -0,0 +1,136 @@
+		t.Fatalf("reuse after Redis went down should skip the ping: created=%v err=%v same=%v", created2, err2, c2 == c1)
+	}
+}
+
+func TestHashPassword(t *testing.T) {
+	if hashRedisPassword("") != "" {
+		t.Error("expected an empty password to hash to empty, not sha256(\"\")")
+	}
+	if hashRedisPassword("secret") == "secret" {
+		t.Error("expected the password to actually be hashed, not passed through")
+	}
+	if hashRedisPassword("secret") != hashRedisPassword("secret") {
```

golangci-lint (2.12.2)

[error] SA4000: identical expressions on the left and right side of the '!=' operator

(staticcheck)